### PR TITLE
Fix merging personalDetails. Use personalDetails for chat message avatars + names.

### PR DESCRIPTION
**Pull request by marcaaron**
cc @NikkiWines 

### Details
At the moment, there is inconsistency between avatars and display names in report comments and those in personal details. This is because report comments come with their own set of avatars/details. This PR does a couple of things:

- Makes it so report comment UI is always referring to the `personalDetails` first instead of report comment data
- Allows merging incomplete sets of personal details into the `formatPersonalDetails` method so that updating a display name won't overwrite the local avatar + vice versa.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/156316

### QA Steps
1. Open a chat
2. Navigate to `/settings/profile` 
3. Update your name
4. Verify it updates in the chat

**Note:** It should not yet update in realtime for any other users. Just the current user.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

File: src/libs/actions/PersonalDetails.js [hunk 1]
```diff
@@ -66,10 +67,6 @@ function getDisplayName(login, personalDetail) {
         return userLogin;
     }
 
-    if (userDetails.displayName) {
-        return userDetails.displayName;
-    }
```
Comment (marcaaron) [inline src/libs/actions/PersonalDetails.js:71]: Not sure if we need this I think this is a generated field anyway and this was preventing us from updating it when the firstName or lastName changes.

**Comment by marcaaron** [comment]
Still need to test this on other platforms but ready for review

**Review by AndrewGable — DISMISSED** [review]

**Comment by marcaaron** [comment]
Tested and building ok on all platforms

**Review by NikkiWines — COMMENTED** [review]
Code _looks_ good but merged master to check how this would work with the edit avatar logic and got some weirdness.

**Master:**
-  Display name takes 30ish seconds to update on the main profile page + chat body

    https://user-images.githubusercontent.com/16747822/113642285-4f6de500-9634-11eb-8ff4-6bb4a1114c2d.mov

**marcaaron-usePersonalDetailsForAvatars + merge master:**
- Seems to remove the avatar (temporarily?) when modifying personal details. Avatar comes back 40ish seconds later.
   - NVP value is unmodified but `myPersonalDetails` and `personalDetails` both have a default avatar temporarily replacing the actual avatar.
- Other personal details update super quickly 
- Profile `Save` button disabling appears to be broken (button should be disabled after pressing `Save` if no additional changes have been made)
    
    https://user-images.githubusercontent.com/16747822/113642168-0d44a380-9634-11eb-854d-9ea60eebc53d.mov

**Comment by marcaaron** [comment]
>NVP value is unmodified but myPersonalDetails and personalDetails both have a default avatar temporarily replacing the actual avatar.

Alright I think I figured this one out.

>Display name takes 30ish seconds to update on the main profile page + chat body

Not seeing this on my local env everything is pretty quick

>Profile Save button disabling appears to be broken (button should be disabled after pressing Save if no additional changes have been made)

I wasn't able to repro this. But let me know if there are more specific steps to try out!

**Review by NikkiWines — DISMISSED** [review]
This looks good. One minor note: I've noticed that it's slow to update the avatar in the chat if you remove the old avatar. But if you upload a new one it updates right away 

https://user-images.githubusercontent.com/16747822/113652302-60285600-9648-11eb-8db5-31ec307e0de8.mov

**Comment by marcaaron** [comment]
>This looks good. One minor note: I've noticed that it's slow to update the avatar in the chat if you remove the old avatar. But if you upload a new one it updates right away

Lemme look into this really quick.

**Comment by marcaaron** [comment]
Ok, so seems like we needed similar logic used for setting the details when we delete the avatar. Created a private helper method for merging the local data after an API call is made to use in both places and I think everything should be working correctly now 😬

**Comment by marcaaron** [comment]
https://user-images.githubusercontent.com/32969087/113656049-bc827a00-9636-11eb-9602-131a0d1681af.mp4

**Review by NikkiWines — APPROVED** [review]
Looks great, works great. :shipit:

**Comment by marcaaron** [comment]
Gonna go ahead and merge this before it gets any conflicts.

**Pull request merged**